src/services/NetworkService.js

```diff
@@ -1,17 +1,66 @@
 import Vue from "vue";
+import promiseRetry from "promise-retry";
+import errcode from "err-code";
 
 const AUTH_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/auth`;
 const API_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/api`;
 const ELIGIBILITY_API_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/eligibility`;
 const CONTACT_API_ROOT = `${process.env.VUE_APP_SERVER_ROOT}/contact`;
 
+const FAULT_TOLERANT_HTTP_TIMEOUT = 10000;
+const FAULT_TOLERANT_HTTP_MAX_RETRY_TIMEOUT = 100000;
+const FAULT_TOLERANT_HTTP_MAX_RETRIES = 10;
+
 export default {
   _successHandler(res) {
     return Promise.resolve(res);
   },
   _errorHandler(res) {
     return Promise.reject(res);
   },
+  _faultTolerantHttp(http, method, onRetry, url, data) {
+    return {
+      isAborted: false,
+      retryHttp: function() {
+        return promiseRetry(
+          retry => {
+            if (this.isAborted) {
+              // early exit
+              return Promise.reject(
+                errcode(new Error("Aborted by user"), "EUSERABORTED")
+              );
+            }
+
+            const promise =
+              ["get", "delete", "head", "jsonp"].indexOf(method) !== -1
+                ? http[method](url, {
+                    timeout: FAULT_TOLERANT_HTTP_TIMEOUT
+                  }).then(this._successHandler, this._errorHandler)
+                : http[method](url, data, {
+                    timeout: FAULT_TOLERANT_HTTP_TIMEOUT
+                  }).then(this._successHandler, this._errorHandler);
+
+            return promise.catch(res => {
+              if (res.status === 0) {
+                if (onRetry) {
+                  onRetry(res, () => {
+                    this.isAborted = true;
+                  });
+                }
+                retry(res);
+              }
+
+              throw res;
+            });
+          },
+          {
+            retries: FAULT_TOLERANT_HTTP_MAX_RETRIES,
+            maxTimeout: FAULT_TOLERANT_HTTP_MAX_RETRY_TIMEOUT
+          }
+        );
+      }
+    }.retryHttp();
```

Comment:
are you creating the retryHttp method and then calling it immediately? unless im readying the brackets wrong, youre doing something like:

```
return {
   isAborted: false,
   retryHttp: function() { ... },
}.retryHttp();
```

so why not just directly call whatever `retryHttp` is doing instead of defining it and then calling it?